### PR TITLE
[RFC] ramips/mt7621: reorganize DTSI files for Netgear R devices

### PR DESCRIPTION
**Pull request by adschm**
This reorganizes DTSI files for the Netgear R devices in mt7621
(and the WNDR3700 v5). I creates a common DTSI for all R devices
and distributes the remaining code in r6220.dtsi to R6220 and
WNDR3700 v5.

### CODE REVIEW

File: target/linux/ramips/dts/mt7621_netgear_r.dtsi [hunk 1]
```diff
@@ -90,18 +64,18 @@
 
 &pcie0 {
 	wifi@0,0 {
-		compatible = "pci14c3,7662";
-		reg = <0x0000 0 0 0 0>;
+		compatible = "mediatek,mt76";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_netgear_r.dtsi]: Can anyone confirm that the generic compatible will work here?

File: target/linux/ramips/dts/mt7621_netgear_r6220.dts [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
 / {
 	compatible = "netgear,r6220", "mediatek,mt7621-soc";
 	model = "Netgear R6220";
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_netgear_r6220.dts]: Despite having an additional button, all Netgear R devices have the same GPIOs for wps and reset, but R6220/WNDR3700v5 are GPIO_ACTIVE_HIGH, while the others are GPIO_ACTIVE_LOW. Is this correct or a mistake?

File: target/linux/ramips/dts/mt7621_netgear_r6220.dts [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
+		reset {
+			label = "reset";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&leds {
+	led_wps: wps {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_netgear_r6220.dts]: led_wps label can be removed in next iteration.

File: target/linux/ramips/dts/mt7621_netgear_wndr3700-v5.dts [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+		reset {
+			label = "reset";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&leds {
+	led_wps: wps {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_netgear_wndr3700-v5.dts]: Remove led_wps label.

**Comment by adrianschmutzler** [comment]
Please propose better names for mt7621_netgear_r.dtsi and mt7621_netgear_r6260_r6350_r6850.dtsi if you have any ideas.

**Comment by adrianschmutzler** [comment]
@achterin @blocktrron @kuyokushin

**Comment by achterin** [comment]
@adrianschmutzler you could name them after their SERCOMM_HWID.
r6260/r6350/r6850: CHJ
r6220: AYA
wndr3700v5: AYB

so maybe mt7621_netgear_r6260_r6350_r6850.dtsi -> mt7621_netgear_chj.dtsi as it is a device family. but i'll leave that up to you cause i think it might cause confusion as we dont use vendor ids for naming thus far.

**Comment by adrianschmutzler** [comment]
@achterin I like that idea. I will use
mt7621_netgear_sercomm.dtsi instead of mt7621_netgear_r.dtsi
and
mt7621_netgear_sercomm_chj.dtsi instead of mt7621_netgear_r6260_r6350_r6850.dtsi

That's not perfect, but way better than before. I don't see less confusion with wndr3700v5 deriving from R6220.dtsi ...

**Comment by adrianschmutzler** [comment]
Updated with sercomm dtsi names.

Would be nice if someone could verify the GPIO_ACTIVE_HIGH vs. GPIO_ACTIVE_LOW (or at least provide a test for a particular device).

File: target/linux/ramips/dts/mt7621_netgear_r.dtsi (hunk 1 above)
Comment (adrianschmutzler) [reply target/linux/ramips/dts/mt7621_netgear_r.dtsi]: @981213 Can you comment on this one?
Some devices have "pci14c3,7662", some have "mediatek,mt76". Can this be merged  as "mediatek,mt76" or do I have to keep separate compatibles?

**Comment by adrianschmutzler** [comment]
Since there have been no further comments, I will merge this in its current state.

**Pull request closed without merging**